### DISCUSSION TITLE
Question about connecting dynamoose with local dynamodb docker

### DISCUSSION THREAD

**Discussion by lokman928** [dynamoose/dynamoose#1525] [Q&A]
Hi all, I am new to dynamoose and I am following the document to let my nestjs application connect to local dynamodb

here is my bootstrap function:
```import { NestFactory } from '@nestjs/core';
import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
import { AppModule } from './app.module';
import * as dynamoose from 'dynamoose';

async function bootstrap() {
    const ddb = new dynamoose.aws.ddb.DynamoDB({
        region: 'us-east-1',
    });
    dynamoose.aws.ddb.set(ddb);
    dynamoose.aws.ddb.local();

    const app = await NestFactory.create(AppModule);

    const config = new DocumentBuilder()
        .setTitle('Tride.id Backend')
        .setVersion('1.0')
        .build();
    const document = SwaggerModule.createDocument(app, config);
    SwaggerModule.setup('api', app, document);

    await app.listen(3000);
}
bootstrap();
```

And when I try to connect the db in my serivce, it said `UnhandledPromiseRejectionWarning: Error: Region is missing`, So is my setting wrong or it is a bug?

**Comment by fishcharlie** [comment]
@Lokman928

From the [documentation](https://dynamoosejs.com/guide/Dynamoose#dynamooseawsddblocalendpoint):

> This function has the same behavior as running dynamoose.aws.ddb.set and passing in an @aws-sdk/client-dynamodb DynamoDB instance with the custom endpoint set. Therefore this function will overwrite any existing custom dynamoose.aws.ddb instance that was set previously.

```js
const ddb = new dynamoose.aws.ddb.DynamoDB({ region: 'us-east-1' });
dynamoose.aws.ddb.set(ddb);
dynamoose.aws.ddb.local();
```
(shortened the code to make it more clear about what lines I'm referring to, exact same code from your original post tho)

Therefore based on that comment in the documentation, lines 1 & 2 of that code snippet won't do anything. Since you immediately overwrite it with `dynamoose.aws.ddb.local`.

I see two possible easy options for how to fix this (there might be more tho):

1. Set the `AWS_REGION` environment variable to your region. This will be used as a fallback if you don't specify one. (https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-region.html#setting-region-environment-variable)
2. Remove `dynamoose.aws.ddb.local();` and set the `endpoint` property in your `dynamoose.aws.ddb.DynamoDB` constructor.

Not sure which you'd prefer. But either will work.

Hope this helps!

**Reply by sulemanelahi** [reply]
@fishcharlie  I'm also facing an issue to connect dynamoose with dynamodb local.

Previously with version 2.7.3 it was working fine


```
if (process.env.AWS_SAM_LOCAL) {
 dynamoose.aws.sdk.config.update({
 region: "local",
 endpoint: "http://host.docker.internal:8000",
 });
}
```

But after upgrading the dynamoose version to 3.1.0 I'm unable to stablish the connection between dynamoose and dynamodb local.

```
dynamoose.aws.ddb.local("http://localhost:8000");
```
Tried this method and also this

```
const ddb = new dynamoose.aws.ddb.DynamoDB({
    "region": "us-east-1"
});

dynamoose.aws.ddb.set(ddb);
```
But not working for connecting with dynamodb local.
Dynamoose try to manipulate the prod table using aws prod/dev credentials.
